asmcomp/arm64/emit.mlp

```diff
@@ -1009,6 +1032,30 @@ let rec emit_all env i =
 
 (* Emission of a function declaration *)
 
+type preproc_fun_result =
+  { max_stack_size : int;
+    contains_nontail_calls : bool }
+
+let preproc_fun env fun_body _fun_name =
+  let rec proc_instr r s i =
+    if i.desc = Lend then r else
+        let upd_size r delta =
+          {r with max_stack_size = max r.max_stack_size (s+delta)}
+        in
+        let (r',s') = match i.desc with
+          | Lop (Istackoffset n) -> (upd_size r n, s+n)
+          | Lpushtrap _ -> (upd_size r 16, s+16)
+          | Lpoptrap -> (r, s-16)
+          | Lop (Icall_ind | Icall_imm _ ) ->
+              ({r with contains_nontail_calls = true}, s)
+          | _ -> (r, s)
```

Comment:
This function should also handle `Ladjust_trap_depth`. 

Rather than a catch-all case, it may be useful to explicitly list all of the variants here so that we can catch a similar bug in the future.

Reply:
Done!